Unit3/Block/User.php

```diff
@@ -0,0 +1,41 @@
+<?php
+declare(strict_types=1);
+
+namespace RLTSquare\Unit3\Block;
+
+use Magento\Framework\View\Element\Template;
+use Magento\Framework\View\Element\Template\Context;
+use RLTSquare\Unit3\Helper\Data;
+
+class User extends Template
+{
+    /**
+     * @var Data
+     */
+    protected Data $helperData;
+
+    public function __construct(Context $context, Data $helperData)
+    {
+        $this->helperData = $helperData;
+        parent::__construct($context);
+    }
+
+    public function unit3Data(): array
```

Comment:
Avoid writing functions like these. It is too obscure, to abstract, the name does not clearly indicate what it should return. There is no documentation over it to show the structure it returns, etc. Break it into several individual functions, e.g. getUsername, getPasword, getEnvironment ... etc... E.g. getUsername and getPassword can then return values based of chosen environemnt ... so then you can have your system.xml store pass and user for each env.